### ISSUE TITLE
nomatch numeric vector should work if destination vector is numeric

### ISSUE THREAD

**Issue by cjyetman** [vincentarelbundock/countrycode#196]
in current master...

```r
countrycode('XXX', 'country.name', 'cown', nomatch = 999)
# [1] NA
# Warning messages:
# 1: In countrycode("XXX", "country.name", "cown", nomatch = 999) :
#   The argument `nomatch` must be NULL, NA, or of the same class
#                 as the destination vector. Filling-in bad matches with NA instead.
# 2: In countrycode("XXX", "country.name", "cown", nomatch = 999) :
#   Some values were not matched unambiguously: XXX
```

**Comment by vincentarelbundock** [comment]
This looks like a simple integer vs numeric class issue:

```
> countrycode('XXX', 'country.name', 'cown', nomatch = as.integer(999))
[1] 999
```

**Comment by vincentarelbundock** [comment]
This check here should be improved: https://github.com/vincentarelbundock/countrycode/blob/master/R/countrycode.R#L207

**Comment by cjyetman** [comment]
Yeah, that whole situation seemed a bit dicey, so I just dealt with the sourcevar as a factor problem for now. I also added a few tests [here](https://github.com/vincentarelbundock/countrycode/blob/master/tests/testthat/test-nomatch.R), which should also be done when this issue gets fixed.

**Comment by vincentarelbundock** [comment]
Sounds great. Thanks!

FYI, I set up Travis for continuous integration. The build badge appears on the README page, so things seem to be working. However, `testthat` was set up mostly for interactive use, and I can't figure out how to have it throw an actual error which Travis' linux virtual machine will pick up. So right now, it shows that the build "passed", even if I included a bad test to intentionally break testthat.
